workflows/devnet-build/Jenkinsfile

```diff
@@ -22,16 +22,35 @@ pipeline {
                     def buildEnv = dir('docker/lisk-build'){
                         return docker.build('lisk-build')
                     }
+                    checkout([$class: 'GitSCM',
+                              branches: [[name: "${params.COMMITISH}" ]],
+                              userRemoteConfigs: [[url: 'https://github.com/LiskHQ/lisk']]])
+                    dir('build') {
+                        sh('''
+                            if [ "x$COMMITSHA" = "xtrue" ]; then
+                               VERSION=$( jq --raw-output '.version' ../package.json )
+                               SHORT_HASH=$( git rev-parse --short HEAD )
+                               OUTPUT_FILE="lisk-$VERSION-$SHORT_HASH-Linux-x86_64.tar.gz"
+                               if s3cmd -q info "s3://lisk-releases/lisk-core/$OUTPUT_FILE" 2> /dev/null ; then
+                                touch .skip_build
```

Comment:
Abort (`currentBuild.rawBuild._this().doTerm();`) the build right here if remote file exists and avoid extra logic later.